### ISSUE TITLE
performance degradation on Oz due to memops

### ISSUE THREAD

**Issue by waterlike86** [emscripten-core/emscripten#15148]
after upgrading from 2.0.15 to 2.0.26 we are seeing a performance degradation due to the memcpy and other memory operations. it seems that it was due to the PR below which moved these operations.
there was talk in the PR about using a JS_MEMOPS flag. is that still open?

https://github.com/emscripten-core/emscripten/pull/13377

**Comment by kripken** [comment]
cc @juj 

This is expected, I think: `-Oz` by definition will do all it can to reduce code size, even make things slower. How does `-Os` compare on your codebase?

**Comment by waterlike86** [comment]
-Oz gives 46.7mb for the main wasm and -Os gives 49.1mb still waiting for performance data..

**Comment by waterlike86** [comment]
looks like theres no real performance difference between Oz and Os, but the 5% size difference impacts startup time by 0.5sec on a slow network.

**Comment by kripken** [comment]
Sounds good, then if there is no performance difference between `-Oz` and `-Os` then `-Oz` is best for your codebase? (and the memops issue is not actually a problem)

**Comment by waterlike86** [comment]
yes Oz is the best choice. I agree that moving memops is not an issue, but would the JS_MEMOPS flag be implemented? on small operations the the speed difference can be quite jarring.

**Comment by kripken** [comment]
I'm not aware of work on `JS_MEMOPS` - that PR was closed and I don't see later references. cc @juj , did you have more thoughts on `JS_MEMOPS` from #13377?
